### PR TITLE
Add a gradle task to generate the UML images

### PR DESCRIPTION
**Pull request by hydraxman**
And generate existing UMLs
image

### CODE REVIEW

File: buildSrc/src/main/java/com/microsoft/hydralab/compile/UMLImageGenerator.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.microsoft.hydralab.compile;
```
Comment (zhou9584) [inline buildSrc/src/main/java/com/microsoft/hydralab/compile/UMLImageGenerator.java:1]: The copyright info is required for each new file.
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

File: agent/doc/UML/test_runner_design.puml [hunk 2]
```diff
@@ -114,7 +114,7 @@ deactivate TaskCompletion
 }
 @endjson
 
-@startuml design
+@startuml TestRunnerListenerDesign
```
Comment (olivershen-wow) [inline agent/doc/UML/test_runner_design.puml:117]: Use underlying pattern to align to the other UMLs' naming.

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -0,0 +1,13 @@
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven {
```
Comment (olivershen-wow) [inline buildSrc/build.gradle:4]: Is this repo used? If not, maybe we should remove it.

File: buildSrc/build.gradle [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        url "https://plugins.gradle.org/m2/"
```
Comment (olivershen-wow) [inline buildSrc/build.gradle:5]: Is this repo used? If not, maybe we should remove it.

File: buildSrc/src/main/java/com/microsoft/hydralab/compile/UMLImageGenerator.java (hunk 1 above)
Comment (hydraxman) [reply buildSrc/src/main/java/com/microsoft/hydralab/compile/UMLImageGenerator.java:1]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package com.microsoft.hydralab.compile;
```

File: agent/doc/UML/test_runner_design.puml (hunk 2 above)
Comment (hydraxman) [reply agent/doc/UML/test_runner_design.puml:117]:
```suggestion
@startuml test_runner_listener_design
```

**Review by olivershen-wow — APPROVED** [review]

**Pull request merged**